### ISSUE TITLE
Additional extensions supported

### ISSUE THREAD

**Issue by gbabiano** [freedomofpress/dangerzone#445]
Could it be possible to extend the support to additional file formats like txt, rtf or html? Thanks

**Comment by apyrgio** [comment]
Plus video and audio formats! Yes, that would be awesome, and we consider this as a future goal, especially if the community is eager for it.

**Comment by apyrgio** [comment]
Adding a link to #163, which refers to `.eml` files.

**Comment by gnoam** [comment]
I would be happy to take a stab at it if possible

**Comment by apyrgio** [comment]
That would be amazing, and we'd be more than happy to help with the implementation. Do you have a specific file type in mind and a suggestion on how to sanitize it?

**Comment by gnoam** [comment]
yeah so i tried with `text/plain` MIME type( txt,csv.tsv) on `dangerzone.conversion.doc_to_pixels.DocumentToPixels.convert` with libreoffice and it seems to do the job. ran the tests and the pdf seem readable.
i need to add another test for encoded files see if works well. i also believe the same approach would  work for rtf. ill set a PR for it asap

if i didnt miss anything i think the next one would be to tackle html and eml. and then maybe use something like puppeteer?

**Comment by deeplow** [comment]
Thanks a lot for that investigation! It sounds pretty plausible to also include rtf. In essence, anything that uses the existing tooling (e.g. libreoffice, `convert`, etc.) and that can be sent as pixels, should be fine.

> if i didnt miss anything i think the next one would be to tackle html and eml. and then maybe use something like puppeteer?

However, here I think we need to careful in the approach. One of our current issues is that we don't want to balloon the container size ([example](https://github.com/freedomofpress/dangerzone/issues/488#issuecomment-1677836861)). The bigger it is, the larger the download sizes. So shipping a full blown web browser may be too much.

But from some experimentation, it looks like `.html` does open with LibreOffice. It's terribly broken, but it does open and one can read most of it.

For `.eml` it does look it's pretty much plain text, so LibreOffice should be fine there.
